### PR TITLE
suppress JCR types from RDF representation

### PR DESCRIPTION
**Pull request by acoburn**
Resolves: https://jira.duraspace.org/browse/FCREPO-1541

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/ManagedRdf.java [hunk 1]
```diff
@@ -35,12 +41,25 @@
 
     private static final Model model = createDefaultModel();
 
+    private static final Set<URI> managedTypes = of(
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/ManagedRdf.java]: Would it be possible to try to reconcile the following as opposed to propagating more flavors of managed and internal and protected, etc:

FedoraTypesUtils.isProtectedAndShouldBeHidden
FedoraTypesUtils.EXPOSED_PROTECTED_JCR_TYPES
FedoraTypesUtils.isSuppressedProperty
FedoraTypesUtils.isInternalProperty
FedoraTypesUtils.isInternalNode

ManagedRdf.isManagedTriple
ManagedRdf.isManagedMixin

RdfLexicon.isManagedPredicate
RdfLexicon.isManagedPredicateURI
RdfLexicon.managedProperties
RdfLexicon.isManagedNamespace
RdfLexicon.managedNamespaces

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 2]
```diff
@@ -403,6 +404,7 @@ public boolean hasType(final String type) {
                 .map(uncheck(x -> x.getName()))
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: Better `uncheck(NodeType::getName)`?

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/ManagedRdf.java (hunk 1 above)
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/ManagedRdf.java]: Do we understand what we mean by "managed" and "internal"? Do those qualities overlap?

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/ManagedRdf.java [hunk 3]
```diff
@@ -35,12 +41,25 @@
...
+            create(MIX_NAMESPACE + "versionable"));
+
     /**
      * No public constructor on utility class
      */
     private ManagedRdf() {
     }
 
+    public static final Predicate<URI> isManagedType =
+        p -> managedTypes.contains(p);
```
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/ManagedRdf.java]: Just `isManagedType = managedTypes::contains`. Do we even need the constant?

**Comment by acoburn** [comment]
This PR has been updated. A question though -- I could have added many of the jcr-related properties and types to `FedoraJcrTypes` (e.g. `mix:created`, `jcr:uuid`, `nt:base`); however, I don't think these values should be defined in the `fcrepo-kernel-api` module at all.

In my opinion, the `FedoraJcrTypes` interface should be renamed to `FedoraTypes` and all of the MODE- and JCR-related properties moved into the `fcrepo-kernel-modeshape` module. But that should be part of a separate PR.

**Comment by ajs6f** [comment]
:+1: to moving all JCR stuff to `-modeshape`.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/0014cd7c6440b5e22112db146970a58c7d623fad

**Pull request closed without merging**